lib/Common/Memory/SmallFinalizableHeapBucket.h

```diff
@@ -174,7 +173,7 @@ class HeapBucketGroup
         typedef typename SmallHeapBlockType<objectAttributes, TBlockAttributes>::BucketType BucketType;
         static BucketType& GetBucket(HeapBucketGroup<TBlockAttributes> * heapBucketGroup)
         {
-            CompileAssert(objectAttributes & FinalizeBit);
+            Assert(objectAttributes & FinalizeBit);
```

Comment:
But the compiler has to know. because to instantiate this function it needs to know the value of objectAttributes already?   If there is an compiler error, you are calling the function with the incorrect template parameter.